src/services/findAllReferences.ts

```diff
@@ -55,43 +33,134 @@ namespace ts.FindAllReferences {
             return undefined;
         }
 
+        const { symbol: aliasedSymbol, shorthandModuleSymbol } = followAliases(symbol, node, typeChecker, isForRename);
+        symbol = aliasedSymbol;
+
+        // Build the set of symbols to search for, initially it has only the current symbol
+        const searchSymbols = populateSearchSymbolSet(symbol, node, typeChecker, implementations);
+        if (shorthandModuleSymbol) {
+            searchSymbols.push(shorthandModuleSymbol);
+        }
+        function isSearchedFor(symbol: Symbol): boolean {

```

Comment:
i would switch this to a helper instead of a parameter and pass in `searchSymbols`.